src/Text/LLVM/AST.hs

```diff
@@ -982,6 +982,10 @@ data ConvOp
   | BitCast
     deriving (Data, Eq, Enum, Generic, Ord, Show, Typeable)
 
+-- | Some of the atomic RW operations are supported only on later LLVM
+-- versions. Because the LLVM version is chosen dynamically, the data
+-- constructors cannot be excluded statically; pretty-printing for an LLVM
+-- version that does not support the operation will call 'error'.
```

Comment:
Really, this caveat applies to many more places than just `AtomicRWOp`, as there are many AST constructs that are only usable on sufficiently recent versions of LLVM. (Thus far, this has only been enforced by writing Haddock comments next to each constructor.)

I propose moving this disclaimer to the module Haddocks and generalizing it a bit. We don't rule out all possible invalid combinations just yet when pretty-printing, but we could at least encourage users to check that what they are attempting to pretty-print is in fact valid on their chosen version of LLVM.